src/Engine/Component/PlayerComponents/PlayerControllerComponent.cpp

```diff
@@ -0,0 +1,84 @@
+#include "PlayerControllerComponent.hpp"
+
+#include "IO/Window.hpp"
+#include "IO/Mouse.hpp"
+#include "IO/Keyboard.hpp"
+#include "Component/SpatialComponents/SpatialComponent.hpp"
+#include "Component/CameraComponents/CameraComponent.hpp"
+#include "Component/SpatialComponents/PhysicsComponents.hpp"
+#include "Component/PostCollisionComponents/GroundComponent.hpp"
+#include "System/SpatialSystem.hpp"
+
+
+
+PlayerControllerComponent::PlayerControllerComponent(float lookSpeed, float moveSpeed, float jumpSpeed) :
+    m_spatial(nullptr),
+    m_newtonian(nullptr),
+    m_camera(nullptr),
+    m_lookSpeed(lookSpeed),
+    m_moveSpeed(moveSpeed),
+    m_jumpSpeed(jumpSpeed),
+    m_enabled(true)
+{}
+
+void PlayerControllerComponent::init(GameObject & go) {
+    Component::init(go);
+    if (!(m_spatial = gameObject()->getComponentByType<SpatialComponent>())) assert(false);
+    if (!(m_newtonian = gameObject()->getComponentByType<NewtonianComponent>())) assert(false);
+    if (!(m_ground = gameObject()->getComponentByType<GroundComponent>())) assert(false);
+    if (!(m_camera = gameObject()->getComponentByType<CameraComponent>())) assert(false);
+}
+
+void PlayerControllerComponent::update(float dt) {
+    if (!m_enabled) {
+        return;
+    }
+    if (Window::isImGuiEnabled()) {
```

Comment:
Why do we need imgui here?

Reply:
keeps the camera from moving around while the mouse moves between imgui panes

Reply:
That should already be the case. The callback functions in Window don't update the Keyboard or Mouse objects if the mouse is hovering over any ImGui panes.

Reply:
right, i mean when imgui is open but the mouse is not over one of the panes. Oftentimes you want to look at something while tweaking imgui, which means you want it in the middle of the screen, which is very hard when moving your mouse from the center of the screen to the imgui pane rotates the camera.

Reply:
I'm still confused. I might just need to look at it after this is merged.